README.md

```diff
@@ -1,6 +1,54 @@
-## GreenX library - prepared within Work Package 2 of the NOMAD Center of Excellence 
+# GreenX Library 
 
-This project is a shared repository for the development of the GreenX library,
-related to work package 2 of the NOMAD Center of Excellence.
+The Green X library is developed under Work Package 2 of the NOMAD Center of Excellence. 
 
-See the GreenX document for the structure of the library (to be made available in due time). The subdirectories reflect in part the structure of the library.
+## Installation
+
+Green X has been designed as a collection of libraries, which can be built relatively 
+independently of one another. To build the whole suite of Green X libraries from the source 
+you need to have a Fortran compiler supporting Fortran 2008, and one of the supported build 
+systems:
+
+* cmake version 3.0.2 or newer, with a build-system backend, i.e. make
+
+**Building with CMake**   
+
+To build all libraries, set up a build directory:
+
+```bash
+mkdir build 
+```
+
+Change to the directory and run cmake configuration:
+
+```bash
+cmake ../
+```
+
+If all requirements are found, build and install the project:
+
+ ```bash
+make -j
+make install 
+ ```
+
+**Running the Tests** 
+
+GreenX uses pytest as its regression testing framework, in conjunction with 
+some custom python modules. First, one must ensure that the python utilities
+are installed. From the GreenX root directory:
+
+```bash
+cd python
+pip install -e .
```

Comment:
I understood that  `--user `mode set the install path to the user install directory, which pertains more to package management rather than sudo privileges? 

I've added a note on using `--user`. As the instructions do not include `sudo`, I think not using it is implied.